controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -384,8 +384,29 @@ var configMapPredicates = predicate.Funcs{
 		if e.ObjectNew.GetName() == "inferenceservice-config" && (namespace == "redhat-ods-applications" || namespace == "opendatahub") { //nolint:goconst
 			return false
 		}
+		// Trigger reconcile function when uninstall configmap has this label
+		operatorNs, _ := cluster.GetOperatorNamespace()
+		if e.ObjectNew.GetNamespace() == operatorNs {
+			labelList := e.ObjectNew.GetLabels()
+			if value, exist := labelList[upgrade.DeleteConfigMapLabel]; exist && value == "true" {
+				return true
+			}
+			return false
+		}
 		return true
 	},
+	CreateFunc: func(e event.CreateEvent) bool {
```

Comment:
this seems to be unrelated right ?

Reply:
i think the main reason to add this new one UninstallconfigMapPredicates is to keep uninstallation with configmap still working if we remove the 0 dsc instance switch-case.

or i misundertsand this comment?

Reply:
I'm not very sure how this is expected to work to be honest as now the watch is configured like:

```go
Watches(
    &corev1.ConfigMap{},
    &handler.EnqueueRequestForObject{},
    builder.WithPredicates(UninstallconfigMapPredicates),
```

Which in my understanding means:
1. `EnqueueRequestForObject` would create a request containing the `Name` and `Namespace` of the object that is the source of the event, so the config map, which does not map to a `DSC`
2. `UninstallconfigMapPredicates` would only validate that source of the event  is a `ConfigMap` in the same namespace as the operator, effectively discarding any other event on any other deployed `ConfigMap`.

So I'm not sure if this is the intended behavior.

Reply:
mh, yep but still the enqueue function is not right, it should always map to a DSC

Reply:
> what i was thinking:
> 
> 1.`Owns(&corev1.ConfigMap{})`is for all the configmap that created by Operator 2. `Watches(&corev1.ConfigMap{},....)` is for the ones that user created by themself.
> 
> when user create a new CM in operator namespace with expected label, or they add expected label to CM. => to trigger DSC reconcile=> into `if upgrade.HasDeleteConfigMap(ctx, r.Client) {...}`

Since `Owns` is a special kind of `Watches` and the result anyway is just triggering of the same `Reconcile` should it be one `Watch`?